Automattic/jetpack
# Social: Render the share status modal globally

We need to render the share status modal globally (with a global state) so that it can be opened/closed from anywhere. The reason for this is that we can trigger the modal from buttons at multiple places like a successful share button, failed share notice retry link etc.

## Proposed changes:

* Create share status modal state (actions, reducer, selectors) in social store
* Create the dedicated `Modal` and `ModalTrigger` components
* Use the unified `ModalTrigger` component in place of existing buttons to fire up the modal
* Create the `GlobalModals` component and render it in Jetpack and Social. We could use the same approach as used for connections management modal, i.e. to render in `PublicizeForm` component, but that didn't work for post publish sidebar because `PublicizeForm` is not rendered in that sidebar.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Publish a post to share it to a few accounts
* Confirm that the post publish share status panel shows the status and the corresponding button
* Click on the button
* Confirm that the modal opens
* Simulate a failure by publishing a post to share to Instagram without a featured image or SIG
* Confirm that you see the failure notice in post publish sidebar
* Confirm that that the link saying "Review status and try again" link opens the modal
* Open Jetpack sidebar for a published post
* Confirm that you see the "Review sharing status" button
* Confirm that the button opens the modal
* Remove the feature flag
* Confirm that you don't see any of the above
